imagePicker.py

```diff
@@ -120,44 +120,47 @@ def GetName(AutoOrCard):
     return None
 
 
+def getCardWithAutocomplete(CardQuery):
+    try:
+        return Autocomplete(q=CardQuery.lower(),
+                            query=CardQuery.lower())
+    except Exception as e:
+        print("Something went wrong. Returning to prompt.")
+        print(f"error details: {e}")
+        print("\n\n")
+        remove('card.png')
+        return False
+
+
+def getCardWithFuzzySearch(CardQuery):
+    CardQuery = Named(fuzzy=CardQuery).name()
+    return getCardWithAutocomplete(CardQuery)
+
+
 def RobustSearch(CardQuery):
     auto = ""
-    try:
-        time.sleep(0.05)
-        card = Named(exact=CardQuery.lower())
-        auto = Autocomplete(q=CardQuery.lower(), query=CardQuery.lower())
-    except Exception:
+    retries = 3
+
+    while retries != 0 or not auto:
         time.sleep(0.05)
         auto = Autocomplete(q=CardQuery.lower(), query=CardQuery.lower())

```

Comment:
I will have to dig in more to Scryfall's API to understand better what could cause autocomplete search failures.  Unless it is a network issue, failing to search by autocomplete once will fail all 3 times, so this will likely only slow things down in the case of not identifying a card on the first try.

Reply:
I also thought about doing the retries *if* there was a TimeOut exception, but didn't find any specific exceptions in the API and didn't have the time to force a timeout during the initialization.

In the API Github page, there's a [https://github.com/NandaScott/Scrython/blob/1484f9f81811a8e9fffa6b21de1ee203a962cc2f/scrython/foundation.py#L5](ScryfallError) exception that you may use, but I'm not entirely sure if it raises only when there's a network issue.

Reply:
Yeah, I looked at that example, and the only exceptions I ever got was errors for not finding cards from a search.  I could dig in to the scryfall API documentation to see if they have a time-out response.  Because if they don't then the time-out exception would be a non-scrython exception and then you could decide to re-try based on that maybe?

Anyways, more research is needed.